File: pset2/vigenere/vigenere.c
#include <string.h>
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

bool allAlpha(string s);

int main(int argc, char** argv)
{
    if (argc != 2 || !allAlpha(argv[1]))
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }

    string k = argv[1];
    int keyLength = strlen(k);
    for (int i = 0, n = strlen(k); i < n; i++)
    {
        if(isupper(k[i]))
        {
            k[i] = k[i] - 'A';
        }

        if(islower(k[i]))
        {
            k[i] = k[i] - 'a';
        }
    }

    string text = get_string("plaintext: ");
    for (int i = 0, n = strlen(text), count = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            if (isupper(text[i]))
            {
                text[i] = (((text[i] - 'A' + k[count]) % 26) + 'A');
            }

            if (islower(text[i]))
            {
                text[i] = (((text[i] - 'a' + k[count]) % 26) + 'a');
            }
            count = (count + 1) % keyLength;
        }
    }

    printf("ciphertext: %s\n", text);
}

bool allAlpha(string s)
{
    for(int i = 0, n = strlen(s); i < n; i++)
    {
        if(!isalpha(s[i]))
        {
            return false;
        }
    }
    return true;
}

File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string, eprint


def main():
    if(len(sys.argv) != 2 or not(allAlpha(sys.argv[1]))):
        eprint("Usage: python vigenere k")
        sys.exit(1)

    k = list(sys.argv[1])
    for i in range(len(k)):
        if (k[i].isupper()):
            k[i] = (ord(k[i]) - ord('A'))
        else:
            k[i] = (ord(k[i]) - ord('a'))

    text = get_string("plaintext: ")
    count = 0
    for i in range(len(text)):
        if(text[i].isalpha()):
            if(text[i].isupper()):
                text = text[:i] + chr(((ord(text[i]) - ord('A') + k[count]) % 26) + ord('A')) + text[i + 1:]
            else:
                text = text[:i] + chr(((ord(text[i]) - ord('a') + k[count]) % 26) + ord('a')) + text[i + 1:]
            count = (count + 1) % len(k)

    print("ciphertext: {0}".format(text))


def allAlpha(s):
    for i in s:
        if(not(i.isalpha())):
            return False
    return True


if __name__ == '__main__':
    main()